pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/ConcurrentMapPartitionUpsertMetadataManager.java

```diff
@@ -288,7 +288,7 @@ protected boolean doAddRecord(MutableSegment segment, RecordInfo recordInfo) {
     // Update metrics
     _serverMetrics.setValueOfPartitionGauge(_tableNameWithType, _partitionId, ServerGauge.UPSERT_PRIMARY_KEYS_COUNT,
         _primaryKeyToRecordLocationMap.size());
-    return !shouldDropRecord.get();
+    return shouldDropRecord.get();
```

Comment:
Suggest still returning `true` when record is added (not out-of-order) to follow common contract of adding value to a collection